server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/AbstractRedirect.java

```diff
@@ -1025,6 +1025,7 @@ protected void updateHeaders() throws MessagingException {
             setIsReply(newMail, isReply(originalMail), originalMail);
 
             newMail.getMessage().saveChanges();
+            newMail.removeAllAttributes();
```

Comment:
It looks like this class is used a lot in james. I can't think of a good way to ensure that this single line of code won't break everything. Have you at least some explanation why it would be a good idea to fix your bug that way ?